esky/patch.py

```diff
@@ -371,20 +371,27 @@ def _write_zipfile_metadata(stream,zfin):
 
 def paths_differ(path1,path2):
     """Check whether two paths differ."""
-    if os.path.isdir(path1):
-        if not os.path.isdir(path2):
-            return True
+
+    return True if not os.path.exists(path1)
+    return True if not os.path.exists(path2)
+
+    isdir1 = os.path.isdir(path1)
+    isdir2 = os.path.isdir(path2)
+    if isdir1 != isdir2:
+        return True
+    
+    if isdir1:
         for nm in os.listdir(path1):
             if paths_differ(os.path.join(path1,nm),os.path.join(path2,nm)):
                 return True
         for nm in os.listdir(path2):
-            if not os.path.exists(os.path.join(path1,nm)):
+            if paths_differ(os.path.join(path1,nm),os.path.join(path2,nm)):
```

Comment:
if you could follow pep8, spaces after commas, the py3k branch is all pep8 and i will cherry pick this over there.